File: frontend/src/lib/ticketing/providers.ts
// TicketProvider — the single frontend source of truth for the
// ASANA/JIRA/GITHUB provider identifier (D-23), matching the backend
// `TicketProvider` str-Enum (backend/app/ticketing/providers.py).
//
// Wire convention (CR-06): uppercase on the wire/backend; PROVIDER_LABELS
// below is the only place this module renders a human-facing label.
export type TicketProvider = 'ASANA' | 'JIRA' | 'GITHUB';

export const PROVIDER_LABELS: Record<TicketProvider, string> = {
  ASANA: 'Asana',
  JIRA: 'Jira',
  GITHUB: 'GitHub',
};

File: backend/app/ticketing/providers.py
"""TicketProvider — the single source of truth for the ASANA/JIRA/GITHUB
provider identifier used across the ticketing subsystem (D-23).

Wire convention (CR-06): stored/compared UPPERCASE on the backend; only
lowercased at the serialization boundary (existing emit sites are
unchanged by this module).
"""

from __future__ import annotations

from enum import Enum


class TicketProvider(str, Enum):
    """Supported ticket-creation providers."""

    ASANA = "ASANA"
    JIRA = "JIRA"
    GITHUB = "GITHUB"
